src/core/Select/Select.tsx

```diff
@@ -322,8 +327,25 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
             </>
           )}
         </div>
-      </DropdownMenu>
-    </div>
+        <span
+          ref={toggleButtonRef}
+          className={cx('iui-end-icon', {
+            'iui-actionable': !disabled,
+            'iui-disabled': disabled,
+            'iui-open': isOpen,
+          })}
+          onClick={() => {
+            if (isOpen) {
+              setIsOpen(false);
+            } else {
+              inputWithIconRef.current?.focus();
+            }
+          }}
```

Comment:
You can replace this whole thing with just `toggle`. I had to use `onPointerDown` because `onClick` was not working for some reason (even with stopPropagation).
```suggestion
        onPointerDown={() => toggle()}
```

Reply:
changed to toggle

Reply:
You might want to try investigating why onClick doesn't work, because onPointerDown is not an ideal solution (it should allow canceling the action if you hold your mouse down, then move mouse away from icon, then release).

Reply:
I think you can just do the same hack as in `Combobox`:
![image](https://user-images.githubusercontent.com/36186912/153160909-a6e83160-c27c-453f-bb40-2be2002362a7.png)
If I remember correctly, the issue here is that when you click on the icon, it is outside click for the `DropdownMenu` and then is like `outside click => close => icon click => open`. Possible fix could be moving `DropdownMenu` to make it as the root element but we moved it in the current place to make it easier to query dropdown in tests. But if we plan to do `appendTo={document.body}` change, then it shouldn't matter as dropdown will be completely separate from `Select`.

Reply:
Currently we moved it down one level to make `focus-within` work.

@gretanausedaite Did you try adding that onClickOutside listener?

If it doesn't work, we should add that focused class in CSS and use the wrapper div as the trigger